### PR TITLE
Fix types and conversions

### PR DESCRIPTION
**Pull request by JonoPrest**
Map the expected types from fields for node js

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -258,73 +260,122 @@ fn map_address_from_binary(v: &Option<FixedSizeData<20>>, should_checksum: bool)
...
+
+fn map_bigint_u128<T: AsRef<[u8]>>(v: &Option<T>) -> Result<Option<BigInt>> {
+    map_opt_result(v.as_ref(), |v| {
+        v.as_ref()
+            .try_into()
+            .context("converting to fixed size bytes u128")
+            .map(u128::from_be_bytes)
+            .map(BigInt::from)
+    })
+}
```
Comment (JonoPrest) [inline src/types.rs:308]: @ozgrakkurt my only question is whether there is any value in mapping to u64 then to BigInt vs u128 then BigInt?

Comment (JonoPrest) [reply src/types.rs:308]: And I would also love it if you could double check the places where I used these two conversions and let me know if you think there is any possibility of overflow. I believe there's enough room though.

File: src/types.rs [hunk 2]
```diff
@@ -258,73 +260,122 @@ fn map_address_from_binary(v: &Option<FixedSizeData<20>>, should_checksum: bool)
...
+fn map_i64_from_u64(v: Option<u64>) -> Result<Option<i64>> {
+    map_opt_result(v, |v| i64::try_from(v).context("converting u64 to i64"))
+}
+
+fn map_i64<T: AsRef<[u8]>>(v: &Option<T>) -> Result<Option<i64>> {
+    map_opt_result(v.as_ref(), |v| {
+        v.as_ref()
+            .try_into()
+            .context("converting to fixed size bytes i64")
+            .map(i64::from_be_bytes)
```
Comment (JonoPrest) [inline src/types.rs:286]: Oh one other thing can you confirm we are using big endian bytes?

File: src/types.rs (hunk 1 above)
Comment (ozgrakkurt) [reply src/types.rs:308]: Ideally want to first use `ruint::aliases::U256::from_be_slice` then use `convert_bigint_signed` and `convert_bigint_unsigned` functions that are already in the library code. Don't need to use u64 or u128

File: src/types.rs (hunk 2 above)
Comment (ozgrakkurt) [reply src/types.rs:286]: can use ruint u256 type and the conversion functions implemented in code already, wrote details in other comment

File: src/types.rs [hunk 3]
```diff
@@ -264,87 +264,63 @@ fn map_hex_string<T: Hex>(v: &Option<T>) -> Option<String> {
...
-fn map_i64<T: AsRef<[u8]>>(v: &Option<T>) -> Result<Option<i64>> {
-    map_opt_result(v.as_ref(), |v| {
-        v.as_ref()
-            .try_into()
-            .context("converting to fixed size bytes i64")
-            .map(i64::from_be_bytes)
+fn map_i64<T: AsRef<[u8]>>(opt: &Option<T>) -> Result<Option<i64>> {
+    map_opt_result(opt.as_ref(), |v| {
+        i64::try_from(ruint::aliases::U256::from_be_slice(v.as_ref()))
+            .context("converting U256 to i64")
```
Comment (JonoPrest) [inline src/types.rs:280]: @ozgrakkurt the only issue with this is it cannot be used for negative number conversions. I don't think cases we are using it for have any negative numbers though so this should be fine.

**Comment by JonoPrest** [comment]
@ozgrakkurt I've updated the map functions

File: src/types.rs (hunk 3 above)
Comment (ozgrakkurt) [reply src/types.rs:280]: If it is negative you can use signed 256bit integer from ruint and use convert_bigint_signed function after

**Review by ozgrakkurt — APPROVED** [review]
LGTM, thanks. Assuming all those numbers are unsigned

**Pull request merged**